jupyter-kernelspec.el

```diff
@@ -71,7 +71,12 @@ information of the host as a prefix."
         (let ((spec-list
                (cdr
                 (split-string
-                 (shell-command-to-string "jupyter kernelspec list")
+                 (if (string-prefix-p "/ssh:" host)
```

Comment:
Actually we don't need this change, the intended usage for remote directories is something like

```emacs-lisp
(let ((default-directory "/ssh:ec2:~"))
  (jupyter-available-kernelspecs))
```

See https://www.gnu.org/software/emacs/manual/html_node/elisp/Magic-File-Names.html, and `process-file` which is what is called by `shell-command-to-string`.

Reply:
oh sorry, I didn't know this and it didn't work for me either (as my local environment and the remote do not have the same login shell and I think `(shell-command-to-string "jupyter kernelspec list")` does not like this).
My local shell is fish and the remote is bash and fish is not even installed there so running 
```
(let ((default-directory "/ssh:fau2:~"))
    (shell-command-to-string "jupyter kernelspec list")))
```
results in
`: /bin/sh: 23: /usr/bin/fish: not found`.
Changing it to
```
(let ((default-directory "/ssh:fau2:~")
        (shell-file-name "/bin/bash"))
    (shell-command-to-string "jupyter kernelspec list")))
```
works.
So now I believe it is an issue with `simple.el` as it uses the `shell-file-name` of the local system even for remote calls.

Reply:
Yes it does look like `shell-command-to-string` is messed up and doesn't handle remote directories properly. I replaced it with a new function `jupyter-command`. If you rebase against the latest master, this problem should be fixed.